diagram.md

```diff
@@ -191,28 +193,39 @@ The `Cluster Elements` contain all of the video frames and audio for each track.
 
-Although the Timecode value is stored once per Cluster, another timecode is stored within the Block structure itself. The way this works is that the Timecode in the Cluster is relative to the entire `Segment`. It is usually the Timecode that the first Block in the Cluster needs to be played at. The Timecode in the Block itself is relative to the Timecode in the Cluster. For example, let's say that the Timecode in the Cluster is set to 10 seconds, and you have a Block in that Cluster that is supposed to be played 12 seconds into the clip; this means that the Timecode in the Block would be set to 2 seconds.
+The `Cues Element` is used to seek when playing back a file by providing a temporal index for each of the `Tracks`. It is similar to the `SeekHead Element`, but used for seeking to a specific time when playing back the file. It is possible to seek without this element, but it is much more difficult because the player has to 'hunt and peck' through the file looking for the correct timecode.
 
-The `ReferenceBlock` in the BlockGroup, is used instead of the basic "P-frame"/"B-frame" description. Instead of simply saying that this Block depends on the Block directly before, or directly afterwards, we put the timecode of the needed Block. And because you can have as many `ReferenceBlock Elements` as you want for a Block, it allows for some extremely complex referencing.
+The `Cues Element` SHOULD contain at least one `CuePoint Element`. Each `CuePoint Element` stores the position of the `Cluster` that contains the `BlockGroup` or `SimpleBlock Element`. The timecode is stored in the `CueTime Element` and location is stored in the `CueTrackPositions Element`.
 
-The `Cues Element` is used to seek when playing back a file by providing a temporal index for each of the tracks. It is similar to the `SeekHead Element`, but this is used for seeking to a specific time when playing back the file. Without this it is possible to seek, but it is much more difficult because the player has to 'hunt and peck' through the file looking for the correct timecode. `Cues` contains `CuePoint Elements` which store the timecode (`CueTime`) and then a listing for the exact position in the file for each of the tracks for that timecode. The `Cues` are pretty flexible for what exactly you want to index. For instance, you can index every single timecode of every `Block` or index selectively. If you have a video file, it is RECOMMENDED to index at least the keyframes of the video track.
+The `Cues Element` is flexible. For instance, `Cues Element` can be used to index every single timecode of every `Block` or can be indexed selectively. For video files, it is RECOMMENDED to index at least the keyframes of the video track.
```

Comment:
I will add it! I don't think of it as being required but it does help with clarity.